std/experimental/allocator/package.d

```diff
@@ -1123,33 +1123,85 @@ allocated with the same allocator.
 */
 void dispose(A, T)(auto ref A alloc, T* p)
 {
+    if (!p) return;
     static if (hasElaborateDestructor!T)
+        p.__dtor;
```

Comment:
I'm not sure, but I think that this will only call the struct destructor, but will not call the destructors of each field after that. See the `destroy` template in object.d for reference.